hail_batch_dev.md

```diff
@@ -3,6 +3,8 @@
 If you only intend to use Hail Batch for running your pipelines, you can skip this
 document. This is meant for developers that work on the Hail Batch codebase.
 
+See [this introduction](https://github.com/hail-is/hail/blob/main/dev-docs/hail-for-new-engineers.md) for a brief history of Hail and an overview of its various components.
```

Comment:
```suggestion
The [Hail dev-docs](https://github.com/hail-is/hail/tree/main/dev-docs) have articles on developer focused guides to Hail. For example, see [this introduction](https://github.com/hail-is/hail/blob/main/dev-docs/hail-for-new-engineers.md) for a brief history of Hail and an overview of its various components.
```

I think broadly introducing the dev docs would be good.

Reply:
👍